R/zchunk_L241.trade_input.R

```diff
@@ -0,0 +1,194 @@
+      write_to_all_regions(c(aglu.NAMES_TECH, "minicam.energy.input", "market.name"), GCAM_region_names) %>%
+      mutate(stub.technology = technology) ->
+      A_demand_technology_R
+    A_an_input_technology %>%
+      write_to_all_regions(c(aglu.NAMES_TECH, "minicam.energy.input", "market.name"), GCAM_region_names) %>%
+      mutate(stub.technology = technology) ->
+      A_an_input_technology_R
+
+    # Add lookup vectors to level1 output tables (55-59)
+    # Add region names to Level1 data tables
+    aglu_demand_calyears <- HISTORICAL_YEARS[HISTORICAL_YEARS %in% MODEL_YEARS]
+    aglu_demand_futureyears <- MODEL_YEARS[!MODEL_YEARS %in% aglu_demand_calyears]
+    L101.ag_kcalg_R_C_Y %>%
+      filter(year %in% aglu_demand_calyears) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      filter(year == max(aglu_demand_calyears)) %>%
+      select(region, GCAM_commodity, value) ->
+      L241.ag_kcalg_R_C_Yf.mlt
+
+    # Fraction of food that should be produced locally
+    LOCAL_FOOD_FRACT <- 0.8
+
+    # Coefficient for ag_trade of food crops (incl secondary products) (61-69)
+    A_demand_technology_R %>%
+      filter(supplysector == "FoodDemand_Crops") %>%
+      left_join_error_no_match(L241.ag_kcalg_R_C_Yf.mlt, by = c("region" = "region", "technology" = "GCAM_commodity")) %>%
+      mutate(coefficient = LOCAL_FOOD_FRACT / round(value, aglu.DIGITS_CALOUTPUT),
+             minicam.energy.input = "ag_trade") %>%
+      select(-value) %>%
+      repeat_add_columns(tibble(year = aglu_demand_futureyears)) %>%
+      select(one_of(aglu.NAMES_STUBTECHCOEF_NM)) ->
+      L241.StubAgTradeCoeff_food
+
+    # Coefficient for ag_trade of non-food crops (incl secondary products) (71-76)
+    A_demand_technology_R %>%
+      filter(supplysector == "NonFoodDemand_Crops") %>%
+      mutate(coefficient = LOCAL_FOOD_FRACT,
+             minicam.energy.input = "ag_trade") %>%
+      repeat_add_columns(tibble(year = aglu_demand_futureyears)) %>%
+      select(one_of(aglu.NAMES_STUBTECHCOEF_NM)) ->
+      L241.StubAgTradeCoeff_nonfood
+
+    # Coefficient for ag_trade of feed crops (incl secondary products) (78-83)
+    A_an_input_technology_R %>%
+      filter(supplysector == "FeedCrops") %>%
+      mutate(coefficient = LOCAL_FOOD_FRACT,
+             minicam.energy.input = "ag_trade") %>%
+      repeat_add_columns(tibble(year = aglu_demand_futureyears)) %>%
+      select(one_of(aglu.NAMES_STUBTECHCOEF_NM)) ->
+      L241.StubAgTradeCoeff_feed
+
+    # Output coefficient for RES ag_trade of food crops (incl secondary products) (85-90)
+    L2012.AgProduction_ag_irr_mgmt %>%
+      filter(year == max(BASE_YEARS)) %>%
+      select(-year) %>%
+      repeat_add_columns(tibble(year = aglu_demand_futureyears)) %>%
+      mutate(res.secondary.output = "ag_trade",
+             output.ratio = 1) %>%
+      select(one_of(aglu.NAMES_AGRES)) ->
+      L241.AgProdTech_RES_output
+
+    # Market for policy portfolio (92-100)
```

Comment:
This sets up the trade regions. For low income regions, we are assuming trade is restricted, so they have to meet the 80% threshold internally. Medium- to High-Income regions have more open trade, so those thresholds can be met anywhere in the region. That is, we assume free trade in medium to high-income regions, and restricted trade in low income regions. Note: these assumptions are derived from the SSP4 storyline (see O'Neill et al., 2017).